### PR TITLE
Hide own tweets & retweets

### PR DESCRIPTION
**Pull request by AlexVanderbist**

### CODE REVIEW

File: resources/assets/js/components/Twitter.vue [hunk 1]
```diff
@@ -76,6 +76,7 @@
                 displayingTopTweetSince: new moment(),
                 onDisplay: [],
                 waitingLine: [],
+                ownScreenName: '@spatie_be'
```
Comment (AlexVanderbist) [inline resources/assets/js/components/Twitter.vue]: @freekmurze can this be hard coded or should I move it to a config file?

File: resources/assets/js/components/Twitter.vue [hunk 2]
```diff
@@ -74,14 +74,14 @@
...
-                onDisplay: [],
+                tweets: [],
                 waitingLine: [],
                 ownScreenName: '@spatie_be'
             };
         },
 
         created() {
-            this.onDisplay = JSON.parse(this.initialTweets).map(tweetProperties => new Tweet(tweetProperties));
+            this.tweets = JSON.parse(this.initialTweets).map(tweetProperties => new Tweet(tweetProperties));
```
Comment (sebastiandedeyne) [inline resources/assets/js/components/Twitter.vue]: Could you look in to removing this `JSON.parse` call while we're at it? Vue should be able to do the parsing for us by declaring the property as a property that needs to be interpreted as JavaScript at the component implementation.

E.g.

```html
<twitter :initial-tweets='{"..":".."}'></twitter>
```

File: resources/assets/js/components/Twitter.vue [hunk 3]
```diff
@@ -110,9 +110,9 @@
                     return;
                 }
 
-                this.onDisplay.unshift(this.waitingLine.shift());
+                this.tweets.unshift(this.waitingLine.shift());
 
-                this.onDisplay = this.onDisplay.slice(0,10);
+                this.tweets = this.tweets.slice(0,10);
```
Comment (sebastiandedeyne) [inline resources/assets/js/components/Twitter.vue]: We could potentially be hiding a lot of tweets, maybe up this to 20? Could be that `initial-tweets` would need an increase too.

File: resources/assets/js/components/Twitter.vue [hunk 4]
```diff
@@ -74,14 +74,14 @@
         data() {
             return {
                 displayingTopTweetSince: new moment(),
```
Comment (sebastiandedeyne) [inline resources/assets/js/components/Twitter.vue]: Can all occurences of `new moment()` be replaced by `moment()`? Feels cleaner to use it as a function.

File: resources/assets/js/components/Twitter.vue (hunk 3 above)
Comment (freekmurze) [inline resources/assets/js/components/Twitter.vue]: 👍

File: resources/assets/js/components/Twitter.vue (hunk 2 above)
Comment (freekmurze) [inline resources/assets/js/components/Twitter.vue]: mmm, I think `JSON.parse` is more readable.

File: resources/assets/js/components/Twitter.vue (hunk 4 above)
Comment (freekmurze) [inline resources/assets/js/components/Twitter.vue]: 👍

File: resources/assets/js/components/Twitter.vue (hunk 2 above)
Comment (sebastiandedeyne) [inline resources/assets/js/components/Twitter.vue]: Conclusion: `this.initialTweets.map(...)` should work without the `JSON.parse` call!

See https://vuejs.org/v2/guide/syntax.html#v-bind-Shorthand

**Comment by sebastiandedeyne** [comment]
🙌

**Pull request merged**